flake.nix

```diff
@@ -1,17 +1,164 @@
 {
   description = "CLI for Hosted Nix binary caches";
-
-  outputs = { self }: let
-    systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
-    forAllSystems = f: builtins.listToAttrs (map (name: { inherit name; value = f name; }) systems);
-  in
-    {
-      packages = forAllSystems (
-        system: {
-          cachix = import ./. { inherit system; };
-        }
-      );
-
-      defaultPackage = forAllSystems (system: self.packages.${system}.cachix);
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/cd8bbdd4fdb15f7758fd3c8df531e9d42bee239d";
+    flake-utils.url = "github:numtide/flake-utils/c0e246b9b83f637f4681389ecabcb2681b4f3af0";
+    # https://discourse.nixos.org/t/recommendations-for-use-of-flakes-input-follows/17413
+    flake-utils.inputs.nixpkgs.follows = "nixpkgs";
+    hls = {
+      url = "github:haskell/haskell-language-server/255498147abb1b18a9cae3c61c5558d7fc84ab31";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    niv = {
+      url = "github:nmattia/niv/82e5cd1ad3c387863f0545d7591512e76ab0fc41";
+      flake = false;
+    };
+    pre-commit-hooks-nix = {
+      url = "github:cachix/pre-commit-hooks.nix/8cb8ea5f1c7bc2984f460587fddd5f2e558f6eb8";
+      inputs.flake-utils.follows = "flake-utils";
+    };
+    gitignore = {
+      url = "github:hercules-ci/gitignore.nix/a20de23b925fd8264fd7fad6454652e142fd7f73";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-compat = {
+      url = "github:edolstra/flake-compat/b4a34015c698c7793d592d66adbab377907a2be8";
+      flake = false;
+    };
+    my-codium = {
```

Comment:
I don't think the editor should be provided since each developer picks their own.